tests/mtd_raw/main.c

```diff
@@ -0,0 +1,463 @@
+    uint32_t page, offset, len;
+
+    if (argc < 5) {
+        printf("usage: %s <dev> <page> <offset> <len>\n", argv[0]);
+        return -1;
+    }
+
+    if (oob) {
+        return -1;
+    }
+
+    page   = atoi(argv[2]);
+    offset = atoi(argv[3]);
+    len    = atoi(argv[4]);
+
+    void *buffer = malloc(len);
+    if (buffer == NULL) {
+        puts("out of memory");
+        return -1;
+    }
+
+    int res = mtd_read_page(dev, buffer, page, offset, len);
+
+    od_hex_dump_ext(buffer, len, 0, page * dev->page_size + offset);
+
+    free(buffer);
+
+    if (res) {
+        printf("error: %i\n", res);
+    }
+
+    return res;
+}
+
+static int cmd_write(int argc, char **argv)
+{
+    bool oob;
+    mtd_dev_t *dev = _get_dev(argc, argv, &oob);
+    uint32_t addr, len;
+
+    if (argc < 4) {
+        printf("usage: %s <dev> <addr> <data>\n", argv[0]);
+        return -1;
+    }
+
+    if (oob) {
+        return -1;
+    }
+
+    addr = atoi(argv[2]);
+    len  = strlen(argv[3]);
+
+    int res = mtd_write(dev, argv[3], addr, len);
+
+    if (res) {
+        printf("error: %i\n", res);
+    }
+
+    return res;
+}
+
+static int cmd_write_page(int argc, char **argv)
+{
+    bool oob;
+    mtd_dev_t *dev = _get_dev(argc, argv, &oob);
+    uint32_t page, offset, len;
+
+    if (argc < 5) {
+        printf("usage: %s <dev> <page> <offset> <len>\n", argv[0]);
+        return -1;
+    }
+
+    if (oob) {
+        return -1;
+    }
+
+    page   = atoi(argv[2]);
+    offset = atoi(argv[3]);
+    len    = strlen(argv[4]);
+
+    int res = mtd_write_page(dev, argv[4], page, offset, len);
+
+    if (res) {
+        printf("error: %i\n", res);
+    }
+
+    return res;
+}
+
+static int cmd_erase(int argc, char **argv)
+{
+    bool oob;
+    mtd_dev_t *dev = _get_dev(argc, argv, &oob);
+    uint32_t addr;
+    uint32_t len;
+
+    if (argc < 4) {
+        printf("usage: %s <dev> <addr> <len>\n", argv[0]);
+        return -1;
+    }
+
+    if (oob) {
+        return -1;
+    }
+
+    addr = atoi(argv[2]);
+    len  = atoi(argv[3]);
+
+    int res = mtd_erase(dev, addr, len);
+
+    if (res) {
+        printf("error: %i\n", res);
+    }
+
+    return res;
+}
+
+static int cmd_erase_sector(int argc, char **argv)
+{
+    bool oob;
+    mtd_dev_t *dev = _get_dev(argc, argv, &oob);
+    uint32_t sector, count = 1;
+
+    if (argc < 3) {
+        printf("usage: %s <dev> <sector> [count]\n", argv[0]);
+        return -1;
+    }
+
+    if (oob) {
+        return -1;
+    }
+
+    sector = atoi(argv[2]);
+
+    if (argc > 3) {
+        count = atoi(argv[3]);
+    }
+
+    int res = mtd_erase_sector(dev, sector, count);
+
+    if (res) {
+        printf("error: %i\n", res);
+    }
+
+    return res;
+}
+
+static void _print_info(mtd_dev_t *dev)
+{
+    printf("sectors: %"PRIu32"\n", dev->sector_count);
+    printf("pages per sector: %"PRIu32"\n", dev->pages_per_sector);
+    printf("page_size: %"PRIu32"\n", dev->page_size);
```

Comment:
The others are without an underscore as well.

```suggestion
    printf("page size: %"PRIu32"\n", dev->page_size);
```